### DISCUSSION TITLE
Kata 2. Acceso denegado en paquetería.

### DISCUSSION THREAD

**Discussion by Gaby-Saldana** [LaunchX-InnovaccionVirtual/CursoIntroPython#155] [Q&A] [answered]
Hola!

Sucede que ya tengo mi entorno virtual creado pero cuando quiero instalar  **python-dateutil** aparece el siguiente mensaje: 
![Acceso denegado](https://user-images.githubusercontent.com/89322665/153687806-b131f561-3670-4516-bd43-a63a92af9d65.png)

¿Me pueden ayudar? ¿Qué podría hacer para que se instale?
También quiero añadir que al momento de crear el entorno las carpetas que se crearon fueron:
-Include
-Lib
-Scripts
y no:
-bin
-include
-lib
No sé si eso tenga algo que ver. 

De antemano gracias!

**Comment by jmonrealr** [comment]
Son por los permisos, trata de abrir `cmd` con permisos de administrador.

Y si no resulta, prueba lo siguiente, igual con permisos de administrador

```Python
py -m pip install mitmproxy
```

**Reply by Gaby-Saldana** [reply]
Aún no :/

Con la primera opción me pidió guardar un archivo .exe,
![1](https://user-images.githubusercontent.com/89322665/153694818-3c20babc-7ecf-4f3f-896a-ecf4520c6503.png)
Lo extraje y era esta terminal:
![2](https://user-images.githubusercontent.com/89322665/153694822-aa855507-2bd9-4378-9247-98d0cb8d3d21.png)
Escribí el comando y actualicé
![2 1](https://user-images.githubusercontent.com/89322665/153694837-c041514d-2fde-4b9e-aa21-5958aa3247db.png)
![2 2](https://user-images.githubusercontent.com/89322665/153694838-f405dbce-76ee-4279-998d-f62e7ac51987.png)
Vuelvo a abrir esa terminal, escribo "pip install python-dateutil"  y ya no tengo que actualizar:
![2 3](https://user-images.githubusercontent.com/89322665/153695052-fad898cb-abdd-4357-b44c-db7c966d1ddb.png)

Pero, ¿y ahora? ¿Sí fue correcto? ¿Hay alguna manera de que esa instalación pueda conectarse a la carpeta en donde está mi entorno? 
Porque entro en la que estoy trabajando y no se puede :/

Y con la segunda opción me apareció esto:
![1 1](https://user-images.githubusercontent.com/89322665/153694931-4a10ebbc-00a7-4614-a804-f62de8385f1d.png)
![1 2](https://user-images.githubusercontent.com/89322665/153694933-0ee79861-ff9e-4c2e-b99f-6ff08d015e9c.png)

Pd. Muchas gracias por la respuesta.

**Reply by jmonrealr** [reply]
A como entendí, te genero un nuevo cmd, pero solo funciona con ese, y con la terminal normal, no te lo permite aun?

**Reply by jmonrealr** [reply]
Puedes probar a volver a generar nuevamente el entorno virtual ó cambiar los permisos de ese directorio.

De igual manera, se podría ejecutar con el `pip install python-dateutil --user` , pero solo se instalaría en teoría para el usuario que se esta usando, pero puede haber problemas con los permisos.

**Reply by Gaby-Saldana** [reply]
Exactamente, generó uno nuevo y ahí pude instalarlo, luego lo intenté en la normal y seguía apareciendo acceso denegado.

Acabo de hacer las otras dos recomendaciones y tampoco :/ ¿Habrá alguna forma de conceder permisos fuera de la terminal?

**Reply by jmonrealr** [reply]
Hmn, con `cmd` no recuerdo bien, te dejo un enlace, que viene con todos los pasos para cambiar el acceso a todos los usuarios, para verificar que si es asi [consulta aqui](https://www.profesionalreview.com/2018/11/24/cambiar-permisos-carpeta/).

Si no es eso, realmente no sabría que pueda llevar a ese error, seria revisar a fondo, o con la comunidad, o en el peor de los casos reinstalar python.

**Reply by Gaby-Saldana** [reply]
Muchas gracias!! ya se pudo con otro comando :D

**Comment by irvingfloley** [comment]
Puedes hacer lo que menciona @jmonrealr en el primer comentario, o puedes probar en la consola de Visual Studio Code a ver si te funciona ahí.

**Reply by Gaby-Saldana** [reply]
Me sigue indicando acceso denegado :/

Gracias por su respuesta :)

**Reply by JosueVerAlar** [reply]
Si tienes una computadora con varios usuarios es posible que sea el problema. Yo apenas estoy empezando y no te puedo explicar mucho esa parte pero los archivos que se crearon son correctos, el archivo "bin" que te solicita la Kata no se genera a veces (depende de varias cosas) pero revisa la carpeta "Scripts", ahí viene el archivo "activate" para activar tu entorno virtual, saludos!

**Reply by Gaby-Saldana** [reply]
@JosueVerAlar Gracias por lo de las carpetas :D

**Comment by AlejandroMontoyaT** [comment, marked as answer]
python3 -m pip install

Ò

python -m pip install

**Reply by Gaby-Saldana** [reply]
Gracias pero aún no!

![3](https://user-images.githubusercontent.com/89322665/153696507-88f2ea89-fbef-4ba6-ae24-288c787bbffa.png)

**Reply by JorgeRodriguez1998** [reply]
Hola! creo que te faltó incluir "python-dateutil" al final de tu linea

Podrías intentar de esta manera a ver si funciona:
python -m pip install python-dateutil

Saludos

**Reply by Gaby-Saldana** [reply]
Sii ya se pudo gracias!!! Después aparecieron otras cosas pero listo, gracias!!!

**Reply by JorgeRodriguez1998** [reply]
Excelenteeeee, ¡mucho éxito!
